meilisearch-release-changelog/v1.9.0.md

```diff
@@ -0,0 +1,215 @@
+
+In order to save storage and run faster, Meilisearch is no longer storing your vector "as-is". Meilisearch now returns the float in a canonicalized representation rather than the user-provided representation.
+
+For example, `3` may be represented as `3.0`
+
+### Document `_vectors` accepts object values
+
+The document `_vectors` field now accepts objects in addition to embedding arrays:
+
+```json
+{
+  "id": 42,
+  "_vectors": {
+    "default": [0.1, 0.2 ],
+      "text": {
+        "embeddings": [[0.1, 0.2, 0.3], [0.4, 0.5, 0.6]],
+        "regenerate": false
+      },
+      "translation": {
+        "embeddings": [0.1, 0.2, 0.3, 0.4],
+        "regenerate": true
+      }
+    }
+}
+```
+
+The `_vectors` object may contain two fields: `embeddings` and `regenerate`.
+
+If present, `embeddings` will replace this document's embeddings.
+
+`regenerate` must be either `true` or `false`. If `regenerate: true`, Meilisearch will overwrite the document embeddings each time the document is updated in the future. If `regenerate: false`, Meilisearch will keep the last provided or generated embeddings even if the document is updated in the future.
+
+This change allows importing embeddings to autoembedders as a one-shot process, by setting them as `regenerate: true`. This change also ensures embeddings are not regenerated when importing a dump created with Meilisearch v1.9.
+
+Meilisearch v1.9.0 also improves performance when indexing and using hybrid search, avoiding useless operations and optimizing the important ones.
+
+## New feature: Ranking score threshold
+
+Use `rankingScoreThreshold` to exclude search results with low ranking scores:
+
+```bash
+curl \
+ -X POST 'http://localhost:7700/indexes/movies/search' \
+ -H 'Content-Type: application/json' \
+ --data-binary '{
+    "q": "Badman dark returns 1",
+    "showRankingScore": true,
+    "limit": 5,
+    "rankingScoreThreshold": 0.2
+ }'
+```
+
+Meilisearch does not return any documents below the configured threshold. Excluded results do not count towards `estimatedTotalHits`, `totalHits`, and facet distribution.
+
+⚠️ For performance reasons, if the number of documents above `rankingScoreThreshold` is higher than `limit`, Meilisearch does not evaluate the ranking score of the remaining documents. Results ranking below the threshold are not immediately removed from the set of candidates. In this case, Meilisearch may overestimate the count of `estimatedTotalHits`, `totalHits` and facet distribution.
+
+Done by @dureuill in #4666
+
+## New feature: Get similar documents endpoint
+
+This release introduces a new AI-powered search feature allowing you to send a document to Meilisearch and receive a list of similar documents in return.
+
+Use the `/indexes/{indexUid}/similar` endpoint to query Meilisearch for related documents:
+
+```sh
+curl \
+  -X POST /indexes/:indexUid/similar
+  -H 'Content-Type: application/json' \
+  --data-binary '{
+    "id": "23",
```

Comment:
What if the ID field name is different, e.g. "productID"?

Reply:
I guess the primary key is used here. @dureuill

Reply:
that's an interesting remark from a user perspective.

The parameter name is `id` we always use the `primaryKey` of the document regardless of its name.
Using `productID` here as the parameter name will not work.

Reply:
Thank you for the clarification!